### ISSUE TITLE
[Bug] ValueError: Invalid element type: <class 'datatree.datatree.DataTree'>

### ISSUE THREAD

**Issue by josenimo** [gustaveroussy/sopa#85]
## Description
I am loading an ome.tif image stack, and trying to segment it. I am following SOPA API tutorial.

## Reproducing the issue
path_to_image ="../991_subset.ome.tif"
sdata = sopa.io.ome_tif(path_to_image, as_image=False)
patches = sopa.segmentation.Patches2D(sdata, element_name='991_subset', patch_width=1200, patch_overlap=50)

``` python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[7], [line 1](vscode-notebook-cell:?execution_count=7&line=1)
----> [1](vscode-notebook-cell:?execution_count=7&line=1) patches = sopa.segmentation.Patches2D(sdata, element_name='991_subset', patch_width=1200, patch_overlap=50)
      [2](vscode-notebook-cell:?execution_count=7&line=2) patches.write()

File /opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:112, in Patches2D.__init__(self, sdata, element_name, patch_width, patch_overlap)
    [110](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:110)     tight, int_coords = True, False
    [111](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:111) else:
--> [112](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:112)     raise ValueError(f"Invalid element type: {type(self.element)}")
    [114](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:114) self.patch_x = Patches1D(xmin, xmax, patch_width, patch_overlap, tight, int_coords)
    [115](https://file+.vscode-resource.vscode-cdn.net/opt/homebrew/Caskroom/mambaforge/base/envs/SOPA/lib/python3.10/site-packages/sopa/patches/patches.py:115) self.patch_y = Patches1D(ymin, ymax, patch_width, patch_overlap, tight, int_coords)

ValueError: Invalid element type: <class 'datatree.datatree.DataTree'>
```

## Expected behavior
I expect patches to be created as geojson shapes

## System
 - OS: MacOS 
 - Python version: 3.10.0
<details>
<summary>Dependencies versions</summary>
<br>

```
# packages in environment at /opt/homebrew/Caskroom/mambaforge/base/envs/SOPA:
#
# Name                    Version                   Build  Channel
aiobotocore               2.11.2                   pypi_0    pypi
aiohttp                   3.9.5                    pypi_0    pypi
aioitertools              0.11.0                   pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
alabaster                 0.7.16                   pypi_0    pypi
anndata                   0.10.8                   pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
app-model                 0.2.7                    pypi_0    pypi
appdirs                   1.4.4                    pypi_0    pypi
appnope                   0.1.4              pyhd8ed1ab_0    conda-forge
array-api-compat          1.7.1                    pypi_0    pypi
asciitree                 0.3.3                    pypi_0    pypi
asttokens                 2.4.1              pyhd8ed1ab_0    conda-forge
async-timeout             4.0.3                    pypi_0    pypi
attrs                     23.2.0                   pypi_0    pypi
babel                     2.15.0                   pypi_0    pypi
botocore                  1.34.19                  pypi_0    pypi
build                     1.2.1                    pypi_0    pypi
bzip2                     1.0.8                h93a5062_5    conda-forge
ca-certificates           2024.7.4             hf0a4a13_0    conda-forge
cachey                    0.2.1                    pypi_0    pypi
cellpose                  3.0.10                   pypi_0    pypi
certifi                   2024.6.2                 pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
cloudpickle               3.0.0                    pypi_0    pypi
colorcet                  3.1.0                    pypi_0    pypi
comm                      0.2.2              pyhd8ed1ab_0    conda-forge
configargparse            1.7                      pypi_0    pypi
connection-pool           0.0.3                    pypi_0    pypi
contourpy                 1.2.1                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
dask                      2024.2.1                 pypi_0    pypi
dask-image                2023.8.1                 pypi_0    pypi
datashader                0.16.2                   pypi_0    pypi
datrie                    0.8.2                    pypi_0    pypi
debugpy                   1.8.2           py310hcf9f62a_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
distributed               2024.2.1                 pypi_0    pypi
docstring-parser          0.16                     pypi_0    pypi
docutils                  0.21.2                   pypi_0    pypi
dpath                     2.2.0                    pypi_0    pypi
exceptiongroup            1.2.1                    pypi_0    pypi
executing                 2.0.1              pyhd8ed1ab_0    conda-forge
fasteners                 0.19                     pypi_0    pypi
fastjsonschema            2.20.0                   pypi_0    pypi
fastremap                 1.14.2                   pypi_0    pypi
fcsparser                 0.2.8                    pypi_0    pypi
filelock                  3.15.4                   pypi_0    pypi
flake8                    7.1.0                    pypi_0    pypi
flexcache                 0.3                      pypi_0    pypi
flexparser                0.3.1                    pypi_0    pypi
fonttools                 4.53.0                   pypi_0    pypi
freetype-py               2.4.0                    pypi_0    pypi
frozenlist                1.4.1                    pypi_0    pypi
fsspec                    2023.6.0                 pypi_0    pypi
geopandas                 1.0.0                    pypi_0    pypi
gitdb                     4.0.11                   pypi_0    pypi
gitpython                 3.1.43                   pypi_0    pypi
h5py                      3.11.0                   pypi_0    pypi
heapdict                  1.0.1                    pypi_0    pypi
hsluv                     5.0.4                    pypi_0    pypi
humanfriendly             10.0                     pypi_0    pypi
idna                      3.7                      pypi_0    pypi
imagecodecs               2024.6.1                 pypi_0    pypi
imageio                   2.34.2                   pypi_0    pypi
imagesize                 1.4.1                    pypi_0    pypi
importlib-metadata        8.0.0              pyha770c72_0    conda-forge
importlib_metadata        8.0.0                hd8ed1ab_0    conda-forge
in-n-out                  0.2.1                    pypi_0    pypi
ipykernel                 6.29.5             pyh57ce528_0    conda-forge
ipython                   8.26.0             pyh707e725_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jinja2                    3.1.4                    pypi_0    pypi
jmespath                  1.0.1                    pypi_0    pypi
joblib                    1.4.2                    pypi_0    pypi
jsonschema                4.22.0                   pypi_0    pypi
jsonschema-specifications 2023.12.1                pypi_0    pypi
jupyter_client            8.6.2              pyhd8ed1ab_0    conda-forge
jupyter_core              5.7.2           py310hbe9552e_0    conda-forge
kiwisolver                1.4.5                    pypi_0    pypi
krb5                      1.21.3               h237132a_0    conda-forge
lamin-utils               0.13.2                   pypi_0    pypi
lazy-loader               0.4                      pypi_0    pypi
legacy-api-wrap           1.4                      pypi_0    pypi
libcxx                    17.0.6               h0812c0d_3    conda-forge
libedit                   3.1.20191231         hc8eb9b7_2    conda-forge
libffi                    3.4.2                h3422bc3_5    conda-forge
libsodium                 1.0.18               h27ca646_1    conda-forge
libsqlite                 3.46.0               hfb93653_0    conda-forge
libzlib                   1.3.1                hfb2fe0b_1    conda-forge
llvmlite                  0.43.0                   pypi_0    pypi
locket                    1.0.0                    pypi_0    pypi
loguru                    0.7.2                    pypi_0    pypi
loompy                    3.0.7                    pypi_0    pypi
magicgui                  0.8.3                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
matplotlib                3.9.0                    pypi_0    pypi
matplotlib-inline         0.1.7              pyhd8ed1ab_0    conda-forge
matplotlib-scalebar       0.8.1                    pypi_0    pypi
mccabe                    0.7.0                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
msgpack                   1.0.8                    pypi_0    pypi
multidict                 6.0.5                    pypi_0    pypi
multipledispatch          1.0.0                    pypi_0    pypi
multiscale-spatial-image  1.0.0                    pypi_0    pypi
napari                    0.4.19.post1             pypi_0    pypi
napari-console            0.0.9                    pypi_0    pypi
napari-matplotlib         2.0.1                    pypi_0    pypi
napari-plugin-engine      0.2.0                    pypi_0    pypi
napari-spatialdata        0.5.0.post0              pypi_0    pypi
napari-svg                0.2.0                    pypi_0    pypi
natsort                   8.4.0                    pypi_0    pypi
nbformat                  5.10.4                   pypi_0    pypi
ncurses                   6.5                  hb89a1cb_0    conda-forge
nest-asyncio              1.6.0              pyhd8ed1ab_0    conda-forge
networkx                  3.3                      pypi_0    pypi
npe2                      0.7.6                    pypi_0    pypi
numba                     0.60.0                   pypi_0    pypi
numcodecs                 0.12.1                   pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
numpy-groupies            0.11.1                   pypi_0    pypi
numpydoc                  1.7.0                    pypi_0    pypi
ome-zarr                  0.9.0                    pypi_0    pypi
opencv-python             4.10.0.84                pypi_0    pypi
opencv-python-headless    4.10.0.84                pypi_0    pypi
openssl                   3.3.1                hfb2fe0b_1    conda-forge
packaging                 24.1               pyhd8ed1ab_0    conda-forge
pandas                    2.2.2                    pypi_0    pypi
param                     2.1.1                    pypi_0    pypi
parso                     0.8.4              pyhd8ed1ab_0    conda-forge
partd                     1.4.2                    pypi_0    pypi
patsy                     0.5.6                    pypi_0    pypi
pexpect                   4.9.0              pyhd8ed1ab_0    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    10.3.0                   pypi_0    pypi
pims                      0.7                      pypi_0    pypi
pint                      0.24.1                   pypi_0    pypi
pip                       24.0               pyhd8ed1ab_0    conda-forge
plac                      1.4.3                    pypi_0    pypi
platformdirs              4.2.2              pyhd8ed1ab_0    conda-forge
pooch                     1.8.2                    pypi_0    pypi
prompt-toolkit            3.0.47             pyha770c72_0    conda-forge
protobuf                  5.27.2                   pypi_0    pypi
psutil                    6.0.0           py310ha6dd24b_0    conda-forge
psygnal                   0.11.1                   pypi_0    pypi
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pulp                      2.7.0                    pypi_0    pypi
pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
pyarrow                   16.1.0                   pypi_0    pypi
pycodestyle               2.12.0                   pypi_0    pypi
pyconify                  0.1.6                    pypi_0    pypi
pyct                      0.5.0                    pypi_0    pypi
pydantic                  2.8.2                    pypi_0    pypi
pydantic-compat           0.1.2                    pypi_0    pypi
pydantic-core             2.20.1                   pypi_0    pypi
pyflakes                  3.2.0                    pypi_0    pypi
pygeos                    0.14                     pypi_0    pypi
pygments                  2.18.0             pyhd8ed1ab_0    conda-forge
pynndescent               0.5.13                   pypi_0    pypi
pyogrio                   0.9.0                    pypi_0    pypi
pyopengl                  3.1.7                    pypi_0    pypi
pyparsing                 3.1.2                    pypi_0    pypi
pyproj                    3.6.1                    pypi_0    pypi
pyproject-hooks           1.1.0                    pypi_0    pypi
pyqt5                     5.15.10                  pypi_0    pypi
pyqt5-qt5                 5.15.14                  pypi_0    pypi
pyqt5-sip                 12.13.0                  pypi_0    pypi
python                    3.10.0          h43b31ca_3_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.10                    4_cp310    conda-forge
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
pyzmq                     26.0.3          py310h16e08c9_0    conda-forge
qtconsole                 5.5.2                    pypi_0    pypi
qtpy                      2.4.1                    pypi_0    pypi
ray                       2.31.0                   pypi_0    pypi
readfcs                   1.1.8                    pypi_0    pypi
readline                  8.2                  h92ec313_1    conda-forge
referencing               0.35.1                   pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
reretry                   0.11.8                   pypi_0    pypi
rich                      13.7.1                   pypi_0    pypi
roifile                   2024.5.24                pypi_0    pypi
rpds-py                   0.18.1                   pypi_0    pypi
s3fs                      0.6.0                    pypi_0    pypi
scanpy                    1.10.2                   pypi_0    pypi
scikit-image              0.24.0                   pypi_0    pypi
scikit-learn              1.5.0                    pypi_0    pypi
scipy                     1.14.0                   pypi_0    pypi
seaborn                   0.13.2                   pypi_0    pypi
session-info              1.0.0                    pypi_0    pypi
setuptools                70.1.1             pyhd8ed1ab_0    conda-forge
shapely                   2.0.4                    pypi_0    pypi
shellingham               1.5.4                    pypi_0    pypi
six                       1.16.0             pyh6c4a22f_0    conda-forge
slicerator                1.1.0                    pypi_0    pypi
smart-open                7.0.4                    pypi_0    pypi
smmap                     5.0.1                    pypi_0    pypi
snakemake                 7.32.4                   pypi_0    pypi
snowballstemmer           2.2.0                    pypi_0    pypi
sopa                      1.1.0                    pypi_0    pypi
sortedcontainers          2.4.0                    pypi_0    pypi
spatial-image             1.1.0                    pypi_0    pypi
spatialdata               0.2.0                    pypi_0    pypi
spatialdata-io            0.1.3                    pypi_0    pypi
spatialdata-plot          0.2.3                    pypi_0    pypi
sphinx                    7.3.7                    pypi_0    pypi
sphinxcontrib-applehelp   1.0.8                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.6                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.5                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.7                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.10                   pypi_0    pypi
sqlite                    3.46.0               h5838104_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
statsmodels               0.14.2                   pypi_0    pypi
stdlib-list               0.10.0                   pypi_0    pypi
stopit                    1.1.2                    pypi_0    pypi
superqt                   0.6.7                    pypi_0    pypi
sympy                     1.12.1                   pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tangram-sc                1.0.4                    pypi_0    pypi
tblib                     3.0.0                    pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
throttler                 1.2.2                    pypi_0    pypi
tifffile                  2024.6.18                pypi_0    pypi
tinycss2                  1.3.0                    pypi_0    pypi
tk                        8.6.13               h5083fa2_1    conda-forge
toml                      0.10.2                   pypi_0    pypi
tomli                     2.0.1                    pypi_0    pypi
tomli-w                   1.0.0                    pypi_0    pypi
toolz                     0.12.1                   pypi_0    pypi
toposort                  1.10                     pypi_0    pypi
torch                     2.3.1                    pypi_0    pypi
tornado                   6.4.1           py310ha6dd24b_0    conda-forge
tqdm                      4.66.4                   pypi_0    pypi
traitlets                 5.14.3             pyhd8ed1ab_0    conda-forge
typer                     0.12.3                   pypi_0    pypi
typing_extensions         4.12.2             pyha770c72_0    conda-forge
tzdata                    2024.1                   pypi_0    pypi
umap-learn                0.5.6                    pypi_0    pypi
urllib3                   2.0.7                    pypi_0    pypi
vispy                     0.14.3                   pypi_0    pypi
wcwidth                   0.2.13             pyhd8ed1ab_0    conda-forge
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
wrapt                     1.16.0                   pypi_0    pypi
xarray                    2024.6.0                 pypi_0    pypi
xarray-dataclasses        1.8.0                    pypi_0    pypi
xarray-datatree           0.0.14                   pypi_0    pypi
xarray-schema             0.0.3                    pypi_0    pypi
xarray-spatial            0.4.0                    pypi_0    pypi
xz                        5.2.6                h57fd34a_0    conda-forge
yarl                      1.9.4                    pypi_0    pypi
yte                       1.5.4                    pypi_0    pypi
zarr                      2.18.2                   pypi_0    pypi
zeromq                    4.3.5                hcc0f68c_4    conda-forge
zict                      3.0.0                    pypi_0    pypi
zipp                      3.19.2             pyhd8ed1ab_0    conda-forge
```

</details>

**Comment by quentinblampey** [comment]
Hello @josenimo, do you have the new version `spatialdata==0.2.0`? It has been released yesterday, and it introduced a few bugs that have been fixed in the `dev` branch of `sopa`. I finalizing some tests, and I will soon release `sopa==1.1.1` which will fix this!

**Comment by josenimo** [comment]
yep just updated as well, I see, thank you for the quick response :) 
Really nice work with SOPA! I am very happy using it. 
Do you recommend any specific multithreading solutions for python API, I was trying out Ray for cellpose segmentation.

**Comment by quentinblampey** [comment]
Regarding multithreading, it really depends on the infrastructure you use. For instance, I'm always using `sopa` on a Slurm HPC, so I run one job per patch, all in parallel (this is what is done in the `sopa` snakemake pipeline by default).
But, of course, I think that Ray is a very good option too!

**Comment by josenimo** [comment]
I am testing on some subsets locally, I will report back on Ray usage :)
Then I will upscale to HPC usage (we use SGE here, so hopefully not too diff)

**Comment by quentinblampey** [comment]
Great, let me know! I'll be happy to test that out, and maybe update the documentation to recommend Ray if it's a good option

**Comment by melonora** [comment]
Hehe yeah sorry for that, what is basically happening is that there are checks for what used to be `SpatialImage` or `MultiscaleSpatialImage`. However these now properly support `xarray.DataArray` and `datatree.DataTree` respectively. For next time

**Comment by melonora** [comment]
I will discuss with Luca next time to do a release candidate so it can be tested on CI beforehand.

**Comment by quentinblampey** [comment]
Yes, I was aware of that, I actually did the change to `xarray.DataArray` and `datatree.DataTree` last week, but didn't have time to make a proper release :)

**Comment by quentinblampey** [comment]
Hi @josenimo, I released a new version, it should be fixed in `sopa==1.1.1`!

**Comment by josenimo** [comment]
great stuff @quentinblampey! I will try it out
